### PR TITLE
[MLC-231] added TestDatasetSize check

### PR DESCRIPTION
**Pull request by yromanyshyn**

### CODE REVIEW

File: deepchecks/checks/methodology/test_dataset_size.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        Args:
+            value (float): minimal allowed test-train ratio.
+
+        Returns:
+            Self: current instance of the TestSize check.
+        """
+
+        def condition(check_result: t.Dict[str, int]) -> ConditionResult:
+            if check_result['train'] < check_result['test']:
+                return ConditionResult(False, 'Train dataset is smaller than test dataset.')
```
Comment (noamzbr) [inline deepchecks/checks/methodology/test_dataset_size.py:82]: That's a separate condition.

File: deepchecks/checks/methodology/test_dataset_size.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                if check_result['test'] <= value
+                else ConditionResult(True)
+            )
+
+        return self.add_condition(
+            name=f'Test dataset size is not smaller than {value}.',
+            condition_func=condition
+        )
+
+    def add_condition_test_train_size_ratio_not_smaller_than(self: TTestSize, value: float = 0.1) -> TTestSize:
```
Comment (noamzbr) [inline deepchecks/checks/methodology/test_dataset_size.py:70]:
```suggestion
    def add_condition_test_train_size_ratio_not_smaller_than(self: TTestSize, ratio: float = 0.01) -> TTestSize:
```

File: deepchecks/checks/methodology/test_dataset_size.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        Dataset.validate_dataset(train, check_name)
+        Dataset.validate_dataset(test, check_name)
+        train.validate_features(check_name)
+        train.validate_label(check_name)
+        test.validate_features(check_name)
+        test.validate_label(check_name)
+        train.validate_shared_features(test, check_name)
+        train.validate_shared_label(test, check_name)
+        return CheckResult(
+            value={'train': train.n_samples, 'test': test.n_samples},
```
Comment (noamzbr) [inline deepchecks/checks/methodology/test_dataset_size.py:45]: display data should be table of the two sizes

File: deepchecks/checks/methodology/test_dataset_size.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                if datasets do not have label column;
+                if datasets do not share same features;
+        """
+        check_name = type(self).__name__
+        Dataset.validate_dataset(train, check_name)
+        Dataset.validate_dataset(test, check_name)
+        train.validate_features(check_name)
+        train.validate_label(check_name)
+        test.validate_features(check_name)
+        test.validate_label(check_name)
```
Comment (noamzbr) [inline deepchecks/checks/methodology/test_dataset_size.py:41]: no need to validate label, shared features

File: deepchecks/checks/methodology/test_dataset_size.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+
+__all__ = ['TestDatasetSize',]
+
+
+TTestSize = t.TypeVar('TTestSize', bound='TestDatasetSize')
+
+
+class TestDatasetSize(SingleDatasetBaseCheck):
+    """Verify Test dataset size."""
```
Comment (noamzbr) [inline deepchecks/checks/methodology/test_dataset_size.py:12]: Should be more detailed - 
Verifies test set size compared to training set size

**Review by noamzbr — COMMENTED** [review]
Added a few comments. Of course:
1. Add notebook
2. Add to Methodology suite
3. In the suite, default conditions would be the ratio one and the one checking that test < train (which will be a separate condition)

File: deepchecks/checks/methodology/test_dataset_size.py (hunk 1 above)
Comment (yromanyshyn) [reply deepchecks/checks/methodology/test_dataset_size.py]: updated

File: deepchecks/checks/methodology/test_dataset_size.py (hunk 2 above)
Comment (yromanyshyn) [reply deepchecks/checks/methodology/test_dataset_size.py]: updated

File: deepchecks/checks/methodology/test_dataset_size.py (hunk 4 above)
Comment (yromanyshyn) [reply deepchecks/checks/methodology/test_dataset_size.py]: updated

File: deepchecks/checks/methodology/test_dataset_size.py (hunk 5 above)
Comment (yromanyshyn) [reply deepchecks/checks/methodology/test_dataset_size.py]: updated

**Review by noamzbr — APPROVED** [review]
